apps/src/blockly/addons/cdoFieldBitmap.ts

```diff
@@ -0,0 +1,36 @@
+import {FieldBitmap} from '@blockly/field-bitmap';
+
+/**
+ * Custom FieldBitmap class with additional hooks for XML serialization.
+ */
+export class CdoFieldBitmap extends FieldBitmap {
+  /**
+   * Constructs a new instance of CdoFieldBitmap.
+   * @param {number[][] | null} value - The initial value of the field, represented as a 2D array of any length, or null/undefined.
+   * @param {object | null} options - The options for the field, can be an object or null/undefined.
+   * @param {object | null} config - Additional configuration options, can be an object or null/undefined.
+   */
+  constructor(value?: number[][], options?: object, config?: object) {
```

Comment:
if you want to be able to pass in null the types here should be `number[][] | null`, etc. Adding `?` allows undefined but not null.

Reply:
Updated here: https://github.com/code-dot-org/code-dot-org/pull/57107